### ISSUE TITLE
[Radio] loses focus when updated on a class component

### ISSUE THREAD

**Issue by slawlertelus** [telus/tds-core#446]
## Description
When using the Radio component from TDS on a class component, and having the Radiobox's value update the class' state, the Radiobox will lose focus due to the class component rerendering. This doesn't appear to happen on non-class components.

## Reproduction Steps
I've cut this code from our repo and modified it for demonstration purposes: https://github.com/telusdigital/my-account-papn/blob/master/ui/src/components/PaymentArrangement/PaymentArrangementPage.jsx#L112

To reproduce, just tab onto a radiobox and press the arrow keys to change the currently selected option. It will change the selection and then cause the radiobox to lose focus (Due to the entire component rerendering).

Please note that I have not tested the code below, I just prepared it to give you an example of how to reproduce. Please ping me (Steven Lawler) or Steve Kotsopoulos on slack for access to our staging environment for a live version if you require it, we have the only account that works.

```

class RadioboxTest extends Component {
  constructor(props) {
    super(props)

    this.state = {
      radioChoice: undefined
    }

    this.setChoice = this.setChoice.bind(this)
  }

  setChoice(event) {
    this.setState({radioChoice: event.target.value})
  }

  render() {
    return (
      <Radio
        label="Choice 1"}
        name="radioboxes"
        value="c1"
        checked={this.state.radioChoice === "c1"}
        onChange={this.setChoice}
      />
      <Radio
        label = "Choice 2"
        name = "radioboxes"
        value = "c2"
        checked = {this.state.radioChoice === "c2"}
        onChange = {this.setChoice}
      />
      <Radio
        label="Choice 3"}
        name="radioboxes"
        value="c3"
        checked={this.state.radioChoice === "c3"}
        onChange={this.setChoice}
      />
    )
  }
}

export default RadioboxTest

```

## Reproduction Percentage (if applicable)

- 10/10 times

## Meta

- [ ] TDS version: v0.32.0

## Screenshots
None

## Description

- Radiobox loses focus when changing value

## Recommendation

- It should keep focus for accessibility

**Comment by theetrain** [comment]
Follow-up: is this issue still unresolved for you? If so, we will re-assess the impact.

**Comment by theetrain** [comment]
@slawlertelus we're closing this issue due to its low priority and low activity. Feel free to add a comment if this is still concerning and we'll gladly re-open this issue.
